backend/inspirehep/search/api.py

```diff
@@ -187,9 +192,29 @@ def execute(self, *args, **kwargs):
 
     def query_for_superuser_or_users(self, query_string):
         if not is_superuser_or_cataloger_logged_in():
-            user_query = Q(
-                IQ(query_string, self) & Q("term", _collections="Literature")
+            non_private_collections = [
+                "Literature",
+                "CDF Notes",
+                "CDS Hidden",
+                "D0 Preliminary Notes",
+                "Fermilab",
+                "HAL Hidden",
+                "LArSoft Notes",
+                "SLAC",
+                "ZEUS Preliminary Notes",
+            ]
+            validation_regex = f"_collections: *({'|'.join(non_private_collections)})"
+            matched_collections = re.match(
+                validation_regex, query_string.lower(), re.IGNORECASE
             )
```

Comment:
1. I'm confused, I thought the idea was to use the `collection` request arg? why are you trying to get the collection from the query string here? I thought the idea was that by default you search in `Literature`, and if a `collection` is passed you search in that collection?
2. Also this is a rather bad way of restricting to allowed collections, as you're relying on one specific spelling of the query, which doesn't necessarily match what the query parser understands. So the correct way to do this is to try to extract the collections from the query parser (which is the thing that knows how to parser a query), but I don't think you actually need to do that due to 1., as it's easy to figure out if a collection is allowed if you don't need to extract it from the query string.

Reply:
No, actually with @drjova we discussed that the first idea is too hackish (from the frontend side) and will require many changes and we decided to do it in a way it's handled on legacy.